src/components/Rate/Rate.tsx

```diff
@@ -0,0 +1,258 @@
+import Rating from "@mui/material/Rating";
+import StarBorder from "@mui/icons-material/StarBorder";
+import FavoriteIcon from "@mui/icons-material/Favorite";
+import FavoriteBorderIcon from "@mui/icons-material/FavoriteBorder";
+import RecordVoiceOverIcon from "@mui/icons-material/RecordVoiceOver";
+import RecordVoiceOverOutlinedIcon from "@mui/icons-material/RecordVoiceOverOutlined";
+import { Button } from "flowbite-react";
+import { useTelegram } from "../UseTelegram";
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { BASE_URL } from "../../Constants";
+import { useParams } from "react-router-dom";
+import {
+  defaultMovie,
+  defaultRate,
+  mockUser,
+} from "../constants";
+import DeleteRateButton from "../DeleteRateButton/DeleteRateButton";
+import { IRate, IMovie } from "../interfaces";
+
+function Rate() {
+  const { movieId } = useParams();
+  const { webApp, executeMethod } = useTelegram();
+  /* const { id, username, first_name } = webApp.initDataUnsafe?.user */
```

Comment:
don't forget to comment values for local testing and uncomment production values